src/Storages/StorageMergeTree.cpp

```diff
@@ -431,6 +431,14 @@ CurrentlyMergingPartsTagger::~CurrentlyMergingPartsTagger()
 
 Int64 StorageMergeTree::startMutation(const MutationCommands & commands, ContextPtr query_context)
 {
+    /// Validate partition IDs (if any) before starting mutation.
+    const auto & set = getPartitionIdsAffectedByCommands(commands, query_context);
+    std::optional<String> partition_id;
+    if (set.size() == 1)
+    {
+        partition_id.emplace(std::move(*set.begin()));
+    }
```

Comment:
else?

Reply:
So for example 3 is not possible?

If it can return empty or single element list - why it return list? May be it should use optional?

Reply:
This is a common code with `StorageReplicatedMergeTree`, we can't change it.

Reply:
Nobody asks to change it. Please add a comment that explains why this function always returns one partition for `StorageMergeTree` (I don't understand why). And throw LOGICAL_ERROR if it returned more than one partition.

Reply:
Please add a comment